packages/react/src/components/Forms/Select/mono-searchable.tsx

```diff
@@ -39,6 +49,99 @@ export const SelectMonoSearchable = forwardRef<SelectHandle, SubProps>(
     const [labelAsPlaceholder, setLabelAsPlaceholder] = useState(
       !downshiftReturn.selectedItem,
     );
+
+    const computeOptionsToDisplay = async (
+      search: string,
+    ): Promise<Option[]> => {
+      const options = await (props.options as CallbackFetchOptions)({ search });
+
+      if (Array.isArray(options)) {
+        setOptionsToDisplay(options);
+      }
+
+      return options;
+    };
+
+    const computeOptionsToDisplayAndDefaultOption = async (
+      defaultValue: Option["value"],
+    ) => {
+      if (defaultValue === undefined) {
+        return;
+      }
+
+      const options = await computeOptionsToDisplay(defaultValue);
+
+      if (Array.isArray(options)) {
+        if (defaultValue) {
+          const defaultOption = options.find(
+            (option) => String(option.value) === defaultValue,
+          );
+
+          if (defaultOption) {
+            downshiftReturn.selectItem(defaultOption);
+          }
+        }
+      }
+    };
+
+    const onInputBlur = () => {
+      setHasInputFocused(false);
+      if (downshiftReturn.selectedItem) {
+        // Here the goal is to make sure that when the input in blurred then the input value
+        // has exactly the selectedItem label. Which is not the case by default.
+        downshiftReturn.selectItem(downshiftReturn.selectedItem);
+      } else {
+        // We want the input to be empty when no item is selected.
+        downshiftReturn.setInputValue("");
+      }
+    };
+
+    const onInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      if (typeof inputProps.onChange === "function") {
+        inputProps.onChange(event);
+      }
+
+      if (isAsyncOptionsFetching) {
+        computeOptionsToDisplay(event.target.value);
+      }
+    };
+
+    const inputProps = downshiftReturn.getInputProps({
+      ref: inputRef,
+      disabled: props.disabled,
+    });
+
+    const renderCustomSelectedOption = !showLabelWhenSelected;
+    const isAsyncOptionsFetching = typeof props.options === "function";
+
+    useEffect(() => {
+      if (!isAsyncOptionsFetching) {
+        return;
+      }
+
+      setIsFetchingInitialOptions(true);
+      let search = props.defaultValue ? String(props.defaultValue) : "";
+
+      // a controlled select can not use props.defaultValue and props.value at the same time
+      if (!props.defaultValue) {
+        search = props.value ? String(props.value) : "";
+      }
+
+      computeOptionsToDisplayAndDefaultOption(search);
+
+      setIsFetchingInitialOptions(false);
+    }, []);
+
+    useEffect(() => {
+      if (isAsyncOptionsFetching) {
+        const toggleMenu = props.isLoading
+          ? () => downshiftReturn.closeMenu()
+          : () => downshiftReturn.openMenu();
+
+        toggleMenu();
+      }
+    }, [props.isLoading]);
+
```

Comment:
This behaviour is weird. Indeed, at the first load, the menu is opened once options are fetched.